packages/design-system/src/components/checkbox.tsx

```diff
@@ -1,86 +1,130 @@
-import React from "react";
-import { styled, CSS, VariantProps } from "../stitches.config";
-import * as CheckboxPrimitive from "@radix-ui/react-checkbox";
-import { CheckIcon } from "@webstudio-is/icons";
-import { theme } from "../stitches.config";
+/**
+ * Implementation of the "Checkbox" component from:
+ * https://www.figma.com/file/sfCE7iLS0k25qCxiifQNLE/%F0%9F%93%9A-Webstudio-Library?node-id=4%3A3352
+ */
 
-const StyledCheckbox = styled(CheckboxPrimitive.Root, {
-  all: "unset",
-  boxSizing: "border-box",
-  userSelect: "none",
-  "&::before": {
-    boxSizing: "border-box",
-  },
-  "&::after": {
-    boxSizing: "border-box",
-  },
+import { forwardRef, type ComponentProps, type Ref } from "react";
+import * as Primitive from "@radix-ui/react-checkbox";
+import { type CSS, css, theme, styled } from "../stitches.config";
+
+const checkedIconStyles = css({
+  display: "none",
+  "[data-state=checked] > &": { display: "block" },
```

Comment:
instead of nesting, better is css custom variable

Reply:
I don't like the boilerplate of variables when encapsulation is not a concern